RELEASE_NOTE_DRAFT.md

```diff
@@ -1,37 +1,108 @@
+* Implemented PrimitiveList.primitiveStream().
+* Implemented PrimitiveMap.updateValues().
+* Implemented factory methods to convert Iterable<BoxedPrimitive> to PrimitiveList, PrimitiveSet, PrimitiveBag, PrimitiveStack.
+* Implemented ofInitialCapacity() and withInitialCapacity() to Primitive Map Factories.
+* Added ability to create ObjectPrimitiveMap, PrimitiveObjectMap, PrimitivePrimitiveMap from Iterable.
+
+# Optimizations
 -----------------
+* Revamped and standardized resize/rehash for all primitive hash structures.
 
-* 
-
-Optimizations
--------------------
-
-* 
-
-Bug Fixes
--------------------
-
-* 
-
-Tech Debt Reduction
--------------------
-
-* 
-
-Library Upgrades
--------------------
-
-* 
+# Bug Fixes
+-----------------
+* Fixed ImmutableDoubletonMap.flipUniqueValues() to throw exception when it has duplicates.
+* Added missing toSentinel() calls in UnifiedSet.put(), UnifiedSetWithHashingStrategy.put().
+* Added validation in FastList for initial capacity < 0.
+
+# Tech Debt Reduction
+---------------------
+* Fixed SerializeTestHelper to use try-with-resources.
+* Reduced duplicate readlock and writelock code in MultiReaderFastList, MultiReaderHashBag and MultiReaderUnifiedSet.
+* Replaced Apache commons-codec by Java 8 Base64 in Verify.
+* Added missing bounded wildcards on generic types.
+* Added wildcard types in MutableMapIterable.putPair(), MutableMapIterable.add(), MutableMultimap.putAllPairs() and MutableMultimap.add().
+* Fixed generics on MultiReaderFastList.newList().
+* Added overrides for methods in MutableOrderedMap.
+* Fixed assignment order in constructor of IntIntervalSpliterator.
+* Removed unnecessary casts, redundant type arguments.
+* Fixed invalid cast in UnifiedSet.trimToSize().
+* Changed Boxed result array to Primitive array to avoid unnecessary boxing in Primitive reduce.
+* Fixed several inspection violations, checkstyle violations and static analysis violations to increase code hygiene. 
+* Fixed incorrect org.eclipse.collections.api.list api doc.
+* Fixed Javadoc lint errors, legacy Javadoc to update references to Eclipse Collections.
+* Moved implementations for forEach(), toSortedListBy(), toSortedSetBy(), toSortedBagsBy(), toSortedMapBy(), toArray(), countBy(), countByWith(), selectUnique(), flatCollectWith(), reverseForEach(), reverseForEachWithIndex() as default methods to interfaces.
+* Moved implementations for tap(), select(), selectWith(), reject(), rejectWith(), collect(), collectWith(), collectIf(), flatCollect(), toImmutable(), toReversed(), reverseThis(), shuffleThis() to MutableList as default implementations.
+* Removed default implementation for RichIterable.toSortedMapBy(), MutableListFactory.ofInitialCapacity(), selectUnique() from all Bag interfaces viz. Bag, ImmutableBag, ImmutableBagIterable, ImmutablePrimitiveBag, ImmutableSortedBag, MutableBag, MutableBagIterable, MutablePrimitiveBag, MutableSortedBag, PrimitiveBag, SortedBag, UnsortedBag.
```

Comment:
duplicated in Breaking changes as well, which makes sense, but just highlighting

Reply:
This is expected, as mentioned in the heading Breaking Changes.